contracts/integrations/test/dydx_integration_test.ts

```diff
@@ -0,0 +1,441 @@
+import { DummyERC20TokenContract, WETH9Contract } from '@0x/contracts-erc20';
+import {
+    blockchainTests,
+    constants,
+    expect,
+    OrderFactory,
+    orderUtils,
+    toHex,
+    TokenBalances,
+} from '@0x/contracts-test-utils';
+import { assetDataUtils, ExchangeRevertErrors } from '@0x/order-utils';
+import { SignedOrder } from '@0x/types';
+import { BigNumber } from '@0x/utils';
+import { TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+
+import { DydxDeploymentManager } from './utils/dydx_deployment_manager';
+
+blockchainTests('dYdX <> 0x integration tests', env => {
+    let owner: string;
+    let solo: string;
+    let untrustedSender: string;
+    let maker: string;
+    let feeRecipient: string;
+    let tradeOriginator: string;
+    let accounts: string[];
+
+    let weth: WETH9Contract;
+    let makerToken: DummyERC20TokenContract;
+    let takerToken: DummyERC20TokenContract;
+    let makerFeeToken: DummyERC20TokenContract;
+    let takerFeeToken: DummyERC20TokenContract;
+
+    let deploymentManager: DydxDeploymentManager;
+    let orderFactory: OrderFactory;
+
+    async function addBalanceAndAllowanceAsync(
+        token: DummyERC20TokenContract | WETH9Contract,
+        tokenOwner: string,
+        tokenSpender: string,
+    ): Promise<void> {
+        if (token instanceof DummyERC20TokenContract) {
+            await token.setBalance.awaitTransactionSuccessAsync(tokenOwner, constants.INITIAL_ERC20_BALANCE, {
+                from: owner,
+            });
+        } else {
+            await token.deposit.awaitTransactionSuccessAsync({
+                from: tokenOwner,
+                value: constants.ONE_ETHER,
+            });
+        }
+        await token.approve.awaitTransactionSuccessAsync(tokenSpender, constants.INITIAL_ERC20_ALLOWANCE, {
+            from: tokenOwner,
+        });
+    }
+
+    before(async () => {
+        [
+            owner,
+            solo,
+            untrustedSender,
+            maker,
+            feeRecipient,
+            tradeOriginator,
+        ] = accounts = await env.getAccountAddressesAsync();
+
+        deploymentManager = await DydxDeploymentManager.deployAsync(env, {
+            owner,
+            trustedMsgSenders: [solo],
+        });
+
+        [makerToken, takerToken, makerFeeToken, takerFeeToken] = deploymentManager.tokens.erc20;
+        weth = deploymentManager.tokens.weth;
+
+        // Configure order defaults
+        const chainId = await env.getChainIdAsync();
+        const defaultOrderParams = {
```

Comment:
If my PR gets merged first, you should check out `AddressManager`, which should eliminate a lot of this code. It might even be useful to extend this to a `dYdXAddressManager`.